### PR TITLE
Enable binary vocab loading from buffer

### PR DESCRIPTION
**Pull request by qianqianzhu**
### Description

This PR enables loading SentencePiece vocabs from the buffer which is required in bergamot-translator.
It is related to https://github.com/browsermt/bergamot-translator/pull/122

List of changes:
- Add `loadFromSerialized()` in `data/vocab.h`, `data/vocab_base.h `, `data/vocab.cpp` and `data/default_vocab.cpp`.
- Implement `loadFromSerialized()` in `data/sentencepiece_vocab.cpp`

Added dependencies: none

### Checklist

- [x] I have tested the code manually
- [ ] I have run regression tests
- [x] I have read and followed CONTRIBUTING.md
- [x] I have updated CHANGELOG.md

### CODE REVIEW

File: src/data/sentencepiece_vocab.cpp [hunk 1]
```diff
@@ -306,6 +306,20 @@ class SentencePieceVocab : public IVocab {
     return spm_->GetPieceSize();
   }
 
+  size_t loadFromSerialized(const string_view& serialized) override {
+    LOG(info, "[data] Loading SentencePiece vocabulary from buffer");
+
+    spm_.reset(new sentencepiece::SentencePieceProcessor());
+    const auto status = spm_->LoadFromSerializedProto(serialized);
```
Comment (jerinphilip) [inline src/data/sentencepiece_vocab.cpp:313]: Can you add a note in comments somewhere on how the mechanics work (I know very little protobuf). Does this expect memory referenced by `string_view` to be alive somewhere, or load a copy and keep it, during when the AlignedMemory that generates it be discarded?

Comment (qianqianzhu) [reply src/data/sentencepiece_vocab.cpp:316]: The note is added.

**Comment by jerinphilip** [comment]
Let's get this in and get bergamot translator submodule to point to master?
/cc @XapaJIaMnu, @kpu

**Comment by XapaJIaMnu** [comment]
@qianqianzhu will this also go upstream, or is there no usecase for it?

**Review by kpu — APPROVED** [review]

**Pull request merged**